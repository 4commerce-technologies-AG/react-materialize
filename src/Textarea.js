import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import idgen from './idgen';
import Icon from './Icon';
import constants from './constants';

class Textarea extends Component {
  componentDidUpdate() {
    if (typeof M !== 'undefined') {
      M.textareaAutoResize(this._textarea);
    }
  }

  renderIcon(icon, className) {
    return (
      <Icon className={cx('material-icons prefix', className)}>{icon}</Icon>
    );
  }

  render() {
    const {
      className,
      s,
      m,
      l,
      xl,
      noLayout,
      iconClassName,
      icon,
      label,
      id,
      onChange,
      ...other
    } = this.props;

    const computedId = id || idgen();

    const sizes = { s, m, l, xl };

    let responsiveClasses;
    if (!noLayout) {
      responsiveClasses = { col: true };
      constants.SIZES.forEach(size => {
        responsiveClasses[size + sizes[size]] = sizes[size];
      });
    }

    const wrapperClasses = cx('input-field', responsiveClasses);

    return (
      <div className={wrapperClasses}>
        {icon && this.renderIcon(icon, iconClassName)}
        <textarea
          {...other}
          ref={input => {
            this._textarea = input;
          }}
          onChange={onChange}
          id={computedId}
          className={cx('materialize-textarea', className)}
        />
        <label htmlFor={computedId}>{label}</label>
      </div>
    );
  }
}

Textarea.propTypes = {
  /*
 * Strip away all layout classes such as col and sX
 */
  noLayout: PropTypes.bool,
  /*
   * Responsive size for Mobile Devices
   */
  s: PropTypes.number,
  /*
   * Responsive size for Tablet Devices
   */
  m: PropTypes.number,
  /*
   * Responsive size for Desktop Devices
   */
  l: PropTypes.number,
  /**
   *  Responsive size for Large Desktop Devices
   */
  xl: PropTypes.number,
  /*
   * render icon next to input
   */
  icon: PropTypes.string,
  /*
   * icon classname
   */
  iconClassName: PropTypes.string,
  /*
   * textarea label
   */
  label: PropTypes.string,
  /*
   * onChange callback
   */
  onChange: PropTypes.func,
  /*
   * override id
   * @default idgen()
   */
  id: PropTypes.string,
  /*
   * disabled input
   */
  disabled: PropTypes.bool,
  /*
  * predefined value
  */
  value: PropTypes.string,
  /*
  * textarea classname
  */
  className: PropTypes.string
};

export default Textarea;
