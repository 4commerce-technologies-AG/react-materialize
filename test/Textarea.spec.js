import React from 'react';
import { shallow } from 'enzyme';
import Textarea from '../src/Textarea';

describe('<Textarea />', () => {
  let wrapper;

  test('Textarea', () => {
    wrapper = shallow(<Textarea />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Disabled', () => {
    wrapper = shallow(<Textarea value="I was here" disabled />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Placeholder', () => {
    wrapper = shallow(<Textarea placeholder="I was here" />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Label', () => {
    wrapper = shallow(<Textarea label="Write something here..." />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Icon', () => {
    wrapper = shallow(<Textarea icon="mode_edit" />);
    expect(wrapper).toMatchSnapshot();
  });
});
