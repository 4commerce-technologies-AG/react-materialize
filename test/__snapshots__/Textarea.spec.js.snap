// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Textarea /> Disabled 1`] = `
<div
  className="input-field col"
>
  <textarea
    className="materialize-textarea"
    disabled={true}
    id="mockId"
    value="I was here"
  />
  <label
    htmlFor="mockId"
  />
</div>
`;

exports[`<Textarea /> Icon 1`] = `
<div
  className="input-field col"
>
  <Icon
    className="material-icons prefix"
  >
    mode_edit
  </Icon>
  <textarea
    className="materialize-textarea"
    id="mockId"
  />
  <label
    htmlFor="mockId"
  />
</div>
`;

exports[`<Textarea /> Label 1`] = `
<div
  className="input-field col"
>
  <textarea
    className="materialize-textarea"
    id="mockId"
  />
  <label
    htmlFor="mockId"
  >
    Write something here...
  </label>
</div>
`;

exports[`<Textarea /> Placeholder 1`] = `
<div
  className="input-field col"
>
  <textarea
    className="materialize-textarea"
    id="mockId"
    placeholder="I was here"
  />
  <label
    htmlFor="mockId"
  />
</div>
`;

exports[`<Textarea /> Textarea 1`] = `
<div
  className="input-field col"
>
  <textarea
    className="materialize-textarea"
    id="mockId"
  />
  <label
    htmlFor="mockId"
  />
</div>
`;
